### PR TITLE
Add static PHP code analysis via psalm

### PR DESCRIPTION
**Pull request by mejo-**
Fixes: #471

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,11 +10,14 @@
 		},
 		"sort-packages": true
 	},
+	"require-dev": {
+		"nextcloud/ocp": "^25.0",
+		"phpunit/phpunit": "^9.5",
+		"psalm/phar": "^5.9"
```
Comment (nickvergessen) [inline composer.json:16]:
```suggestion
		"vimeo/psalm": "^5.9"
```

File: psalm.xml
Comment (nickvergessen) [inline psalm.xml:1]: Is the file already excluded via the Makefile?

Comment (mejo-) [reply psalm.xml:1]: Good catch :ok_hand:

**Review by nickvergessen — COMMENTED** [review]
Should also add https://github.com/nextcloud/.github/blob/master/workflow-templates/update-nextcloud-ocp-matrix.yml then (with stable25 since that is the oldest version supported by info.xml)

File: composer.json (hunk 1 above)
Comment (mejo-) [reply composer.json:16]: What's the difference and why is `vimeo/psalm` superiour?

Comment (nickvergessen) [reply composer.json:16]: It's the original source and I think the phar thing is heading to the grave

File: tests/psalm-baseline.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<files psalm-version="5.9.0@8b9ad1eb9e8b7d3101f949291da2b9f7767cd163">
+  <file src="vendor/nextcloud/ocp/OCP/BackgroundJob/TimedJob.php">
```
[the remark this answers is not in the document]
Comment (mejo-) [reply tests/psalm-baseline.xml:3]: Well, updating the baseline always adds it :thinking: It's also in the baseline of Text for example.

File: psalm.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+>
+	<stubs>
+		<file name="tests/stub.phpstub" preloadClasses="true"/>
+	</stubs>
+    <projectFiles>
+        <directory name="lib" />
+    </projectFiles>
+    <extraFiles>
+        <directory name="vendor/nextcloud/ocp" />
+    </extraFiles>
```
[the remark this answers is not in the document]
Comment (mejo-) [reply psalm.xml]: Ah, that's very helpful! Changed now.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**